homeUi/src/test/kotlin/com/gravatar/app/homeUi/presentation/home/profile/ProfileViewModelTest.kt

```diff
@@ -0,0 +1,109 @@
+package com.gravatar.app.homeUi.presentation.home.profile
+
+import app.cash.turbine.test
+import com.gravatar.app.homeUi.presentation.home.profile.about.AboutInputField
+import com.gravatar.app.testUtils.CoroutineTestRule
+import com.gravatar.restapi.models.Profile
+import com.gravatar.services.GravatarResult
+import com.gravatar.services.ProfileService
+import io.mockk.coEvery
+import io.mockk.mockk
+import kotlinx.coroutines.test.StandardTestDispatcher
+import kotlinx.coroutines.test.runTest
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertTrue
+import org.junit.Rule
+import org.junit.Test
+import java.net.URI
+
+class ProfileViewModelTest {
+    private val testDispatcher = StandardTestDispatcher()
+
+    @get:Rule
+    var coroutineTestRule = CoroutineTestRule(testDispatcher)
+
+    private val profileService = mockk<ProfileService>(relaxed = true)
+
+    private lateinit var viewModel: ProfileViewModel
+
+    @Test
+    fun `when viewmodel is initialized and fetch profile finishes successfully then uiState contains profile`() =
+        runTest {
+            val mockProfile = mockProfile()
+
+            coEvery {
+                profileService.retrieveCatching(any<String>())
+            } returns GravatarResult.Success(mockProfile)
+
+            viewModel = initViewModel()
+
+            viewModel.uiState.test {
+                assertEquals(ProfileUiState(isLoading = false), awaitItem())
+                assertEquals(ProfileUiState(isLoading = true), awaitItem())
+                assertEquals(ProfileUiState(isLoading = false, profile = mockProfile), awaitItem())
+                assertEquals(
+                    ProfileUiState(isLoading = false, profile = mockProfile, aboutFields = mockProfile.aboutFields()),
+                    awaitItem()
+                )
+            }
+        }
+
+    @Test
+    fun `when profile is available then aboutFields are correctly populated`() {
+        // Given
+        val mockProfile = mockProfile()
```

Comment:
```suggestion
        val profile = defaultProfile()
```

Could we rename? This is not a mock and it confused me a bit :D